src/target/riscv/riscv-013.c

```diff
@@ -3305,19 +3330,7 @@ int riscv013_dmi_write_u64_bits(struct target *target)
 
 static int maybe_execute_fence_i(struct target *target)
 {
-	RISCV013_INFO(info);
-	RISCV_INFO(r);
-	if (info->progbufsize + r->impebreak >= 2) {
```

Comment:
By removing this check, you're always reporting an error on debug systems where you can't execute a fence, eg. because they don't have a Program Buffer. They already get a warning about this at https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/riscv-013.c#L1443.

Reply:
IIRC I removed this warning because it was firing on systems I knew there was a progbuf and I just didn't want to deal with it at the time, but I might be wrong about that.

Reply:
Let's leave this check, but inside of it call execute_fence() to get the new ultra-paranoid behavior.